src/Storage/Notmuch.hs

```diff
@@ -60,21 +61,21 @@ applyTagOp (RemoveTag t) = removeTags [t]
 applyTagOp ResetTags = set tags []
 
 class ManageTags a  where
-    tags :: Lens' a [T.Text]
+    tags :: Lens' a [Tag]
 
-setTags :: (ManageTags a) => [T.Text] -> a -> a
+setTags :: (ManageTags a) => [Tag] -> a -> a
```

Comment:
Aaah of course. Yes... really good idea to change it straight into a `Notmuch.Tag`. Cheers!